src/lib/resolvers/enum.ts

```diff
@@ -0,0 +1,21 @@
+import { Identifiers } from '../errors/Identifiers';
+import { err, ok, Result } from '../parsers/Result';
+
+export function resolveEnum(
+	parameter: string,
+	options?: { enum?: string[]; caseInsensitive?: boolean }
```

Comment:
what's the reasoning for having `options.enum` be nullable?

Reply:
Sapphire argument stuff

Reply:
To have a consistency that also the options of the argument can be nullable.